staging/src/k8s.io/kms/encryption/service.go

```diff
@@ -0,0 +1,261 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encryption
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	aestransformer "k8s.io/kms/pkg/encrypt/aes"
+	"k8s.io/kms/pkg/value"
+	"k8s.io/kms/service"
+	"k8s.io/utils/lru"
+)
+
+var (
+	// emptyContext is an empty slice of bytes. This is passed as value.Context to the
+	// GCM transformer. The grpc interface does not provide any additional authenticated data
+	// to use with AEAD.
+	emptyContext = value.DefaultContext([]byte{})
+	// errInvalidKMSAnnotationKeySuffix is returned when the annotation key suffix is not allowed.
+	errInvalidKMSAnnotationKeySuffix = fmt.Errorf("annotation keys are not allowed to use %s", referenceSuffix)
+
+	localKEKGenerationPollInterval = 1 * time.Second
+	localKEKGenerationPollTimeout  = 5 * time.Minute
```

Comment:
```suggestion

	// these are var instead of const so that we can set them during tests
	localKEKGenerationPollInterval = 1 * time.Second
	localKEKGenerationPollTimeout  = 5 * time.Minute
```